lib/ansible_tower_client/base_models/workflow_job_node.rb

```diff
@@ -5,14 +5,20 @@ def workflow_job
     end
 
     def job
-      api.jobs.find(job_id) if job?
+      return unless job?
+
+      if related.job.include?("/jobs/")
+        api.jobs.find(job_id)
+      elsif related.job.include?("/workflow_jobs/")
+        api.workflow_jobs.find(job_id)
+      end
```

Comment:
Unsure how best to handle this, should `job` return `Job` and `WorkflowJob` types?  That would match the behavior of the tower API where there is a workflow_job id in the `job_id` property.  @bdunne opinions?